mantidproject/mantid
# ISIS Powder Pearl added default Rebin value

**Description of work.**
Added a parameter `focused_bin_widths parameter` in` pearl_advanced_config.py` to allow setting of default values to rebin pearl focus' to.

**To test:**
(this test requires access to the ISIS Data Search Archive to access some of the files)
Run the following script with these [files](https://github.com/mantidproject/mantid/files/2495235/PEARL.zip)   (note some additional files are required, but they were to large to transfer via github, please message me on slack to receive them)
```
from isis_powder import Pearl

config_file_path = r"/Path/to/pearl_shared_config.yaml"
Pearl_example = Pearl(config_file=config_file_path,user_name="testold70")
Pearl_example.create_vanadium(run_in_cycle=101508, do_absorb_corrections=True)
AnalysisDataService.clear()
Pearl_example.focus(run_number="101508-101515", focus_mode="all")
```
you will need to update the paths both in the script and in pearl_shared_config.yaml
Open up the workspace group ResultsD and check the history of any of the workspaces it contains, ~24 entries from the bottom should be a group of Rebins, all of these rebins should use -0.0006 as there rebin stepsize, then close mantid and adjust any of the numbers within `focused_bin_widths` in the `pearl_advanced_config.py` in `scripts/Diffraction/isis_powder/pearl_routines/`restart mantid and carry out the test again, the numbers you changed should lead to those rebins using different values.

Re #23092


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
